src/properties.jl

```diff
@@ -246,39 +246,42 @@ rmatrices = Dict(
 
 #---------------------- Equivariant matrices
 
-struct EuclideanMatrix{T} <: AbstractProperty where T<:Real
-   val::SMatrix{3, 3, Complex{T}, 9}
+struct EuclideanMatrix{T} <: AbstractProperty
+   val::SMatrix{3, 3, T, 9}
 end
 
 Base.show(io::IO, φ::EuclideanMatrix) = 
    print(io,  "m[$(φ.val[1,1]), $(φ.val[1,2]), $(φ.val[1,3])]
 m[$(φ.val[2,1]), $(φ.val[2,2]), $(φ.val[2,3])] 
 m[$(φ.val[3,1]), $(φ.val[3,2]), $(φ.val[3,3])]")
 
-real(φ::EuclideanMatrix) = EuclideanMatrix(φ.val)
-complex(φ::EuclideanMatrix) = EuclideanMatrix(φ.val)
+real(φ::EuclideanMatrix) = EuclideanMatrix(real.(φ.val))
+complex(φ::EuclideanMatrix) = EuclideanMatrix(complex(φ.val))
 complex(::Type{EuclideanMatrix{T}}) where {T} = EuclideanMatrix{complex(T)}
 
 +(x::SMatrix{3}, y::EuclideanMatrix) = EuclideanMatrix(x + y.val)
 Base.convert(::Type{SMatrix{3, 3, T, 9}}, φ::EuclideanMatrix) where {T} =  convert(SMatrix{3, 3, T, 9}, φ.val)
```

Comment:
where is this needed?

Reply:
not sure, I kind of copied the implementations of `EuclideanVector` here... Should I check, or should we introduce a new abstract type to avoid code redundancy?

Reply:
ok, leave it for now.